### PR TITLE
AF-2509 Dart2

### PR DESCRIPTION
**Pull request by robbecker-wf**
# Summary
Fixes to allow running under Dart 2

# Changes
- Removed use of dart_dev since it wasn't actually being run anywhere
- Removed use of dart_dev in test fixtures since dart_dev still doesn't pub get under Dart 2, so was causing tests to fail. Also, I didn't like the fact that dart_dev could update and cause tests to fail here. So, I provided an in repo fake project to replace it.
- Run travis tests in Dart 1.24.3 & Dart 2
- Fixed an oversight when running dart fmt check on travis that didn't set the exit code, so it wouldn't ever fail the build if formatting was needed

# Testing
 - [x] CI passes
 - [x] Smoke test on your favorite repos if you like

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

File: .travis.yml [hunk 1]
```diff
@@ -2,10 +2,11 @@ language: dart
 
 dart:
   - 1.24.3
+  - 2.0.0
 
 script:
   - dartanalyzer .
-  - pub run dart_style:format -l 120 -n .
+  - pub run dart_style:format -l 120 -n --set-exit-if-changed .
```
Comment (robbecker-wf) [inline .travis.yml]: need to have dart format set the exit code if you want to fail the build

File: test/executable_test.dart [hunk 2]
```diff
@@ -136,7 +136,7 @@ void main() {
         expect(result.exitCode, 1);
         expect(result.stderr,
             contains('These packages may be unused, or you may be using executables or assets from these packages:'));
-        expect(result.stderr, contains('dart_dev'));
+        expect(result.stderr, contains('fake_project'));
```
Comment (robbecker-wf) [inline test/executable_test.dart]: replaced dart_dev in tests with fake_project. You'll see a lot of this.

File: test_fixtures/analyzer/pubspec.yaml [hunk 3]
```diff
@@ -1,10 +1,10 @@
 name: analyzer_dep
 version: 0.0.0
 private: true
-
+environment:
+  sdk: '>=1.24.2 <3.0.0'
```
Comment (robbecker-wf) [inline test_fixtures/analyzer/pubspec.yaml]: Let the test fixture projects install under Dart 2. Also a lot of these.

**Comment by robbecker-wf** [comment]
@Workiva/release-management-pp

**Comment by evanweible-wf** [comment]
QA +1
- [x] CI passes on Dart 1 and Dart 2

**Pull request merged**